### ISSUE TITLE
Add `cmdix` as (optional?) replacement for `xoreutils`

### ISSUE THREAD

**Issue by gforsyth** [xonsh/xonsh#3178]
As per some discussion in #3057:

Or if you like your commands implemented in Python, consider [cmdix](https://pypi.org/project/cmdix), a pure-python implementation of the gnu coreutils.

_Originally posted by @jaraco in https://github.com/xonsh/xonsh/issues/3057#issuecomment-482739104_

We can use this in place of `xoreutils` since why duplicate effort?

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by scopatz** [comment]
I am 100% in favor of this

**Comment by jnoortheen** [comment]
does this mean updating `setup.py` and remove xoreutils from xonsh ?

**Comment by gforsyth** [comment]
Yeah, I would make `cmdix` a gated import, add it as an optional dependency to the `setup.py` and then have a xontrib that loads up those commands

**Comment by jnoortheen** [comment]
Ok. Thanks for clarifying. I occasionally use windows and was searching for something like this.

**Comment by jnoortheen** [comment]
I have checked the [cmdix](https://github.com/jaraco/cmdix) project, it seems incomplete as commands like [cat]() will not be able to handle stdin . 

There are projects like below provide better binaries, that is performant. 
  - https://github.com/uutils/coreutils (haven't used it, but gives more number of commands including grep)
  - http://gnuwin32.sourceforge.net/packages/coreutils.htm   -- (`scoop install coreutils`)

<img width="576" alt="image" src="https://user-images.githubusercontent.com/6702219/160232582-a055ebad-5a7d-4673-b3a8-519ba8b68660.png">
  
Also I don't think Python will be good for implementing such functionality. I suggest moving some of the xoreutils out and keep only the necessary ones within the xonsh package.

**Comment by anki-code** [comment]
This kind of things should be installed as a xontrib so new xontribs are welcome and I'm going to close this.
